account_credit_control_queue_job/models/credit_control_communication.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2025 360ERP (<https://www.360erp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.tools import split_every
+
+
+class CreditControlCommunication(models.Model):
+    _inherit = "credit.control.communication"
+
+    def _send_mails(self):
+        key = "account_credit_control_queue_job.batch_size"
+        batch_size = self.env["ir.config_parameter"].sudo().get_param(key)
+        try:
+            batch_size = max(1, int(batch_size))
+        except Exception:  # pylint: disable=broad-except
+            batch_size = 1
+        for comms in split_every(batch_size, self.ids, self.browse):
+            if batch_size > 1:
+                desc = _("Sending credit control emails for ids: %s") % comms.ids
+            else:
+                desc = _("Sending credit control email for %s") % comms.partner_id.name
+            comms.with_delay(description=desc)._send_communications_by_email()
```

Comment:
Sorry for being unclear, by "batch" I meant `queue.job.batch`. It is one job per communication but they are grouped in batches.

https://github.com/OCA/credit-control/blob/12.0/account_credit_control_queue_job/wizard/credit_control_emailer.py#L17-L36

Reply:
All right, done in https://github.com/OCA/credit-control/pull/467/commits/ae9d77b43b9918cd384f8263e3784bc5b2ac3881
I had seen some other custom implementation of the feature use such a batch size for the email sending, so I thought that's what you meant.